packages/terra-cli/index.js

```diff
@@ -0,0 +1,78 @@
+const yargs = require('yargs/yargs');
+const path = require('path');
+const fs = require('fs-extra');
+
+const Logger = require('./lib/utils/Logger');
+
+const setupCLI = () => {
+  const cli = yargs();
+
+  return cli.usage('Usage: $0 <command> [options]')
+    .demandCommand(1, 'A command is required. Pass --help to see all available commands and options.')
+    .recommendCommands()
+    .strict()
+    .fail((msg, err) => {
+      // certain yargs validations throw strings
+      const actual = err || new Error(msg);
+
+      // ValidationErrors are already logged, as are package errors
+      if (actual.name !== 'ValidationError' && !actual.pkg) {
+        // the recommendCommands() message is too terse
+        if (/Did you mean/.test(actual.message)) {
+          // eslint-disable-next-line no-console
+          console.error(`Unknown command "${cli.parsed.argv._[0]}"`);
+        }
+
+        // eslint-disable-next-line no-console
+        console.error(actual.message);
+      }
+
+      // exit non-zero so the CLI can be usefully chained
+      cli.exit(actual.code > 0 ? actual.code : 1, actual);
+    })
+    .alias('h', 'help')
+    .alias('v', 'version')
+    .wrap(cli.terminalWidth());
+};
+
+const subDirectoriesOfDirectory = (directoryPath) => (
+  fs.readdirSync(directoryPath, { withFileTypes: true })
+    .filter(directoryEntry => directoryEntry.isDirectory())
+    .map(directoryEntry => path.join(directoryPath, directoryEntry.name))
+);
+
+const terraClIDirectoriesFromPaths = (paths, preferSrc) => (
+  paths.map(dependencyPath => path.join(dependencyPath, preferSrc ? 'src' : 'lib', 'terra-cli')).filter(directory => fs.pathExistsSync(directory))
+);
+
+const loadTerraCommands = () => {
+  const { dependencies = [], devDependencies = [], name } = fs.readJSONSync(path.join(process.cwd(), 'package.json'));
+  const firstLevelDependencies = [...Object.keys(dependencies), ...Object.keys(devDependencies)];
+  const firstLevelDependencyPaths = firstLevelDependencies.map(directory => path.join(process.cwd(), 'node_modules', directory));
```

Comment:
This will allow anyone to hook in for a command correct, if they add a terra-cli directory to their project?

Reply:
We should impose tight control over what commands can be added to terra-cli. Any command in terra cli should be supported and maintained by the terra team.

Reply:
I'm fine with that.  I misunderstood things coming out of the war room.  I had it documented as above on the design, but updated it to be (additions in bold):

To cover all cases, we will search the following **for a list of projects that we control in an allow list**

Reply:
Updated to behave as mentioned: https://github.com/cerner/terra-toolkit/pull/460/commits/6c35f4191a2115b9652c22dc9f42105554d0b08c